src/main/java/com/mineboundteam/minebound/inventory/ArmorForgeMenu.java

```diff
@@ -0,0 +1,153 @@
+package com.mineboundteam.minebound.inventory;
+
+import com.mineboundteam.minebound.crafting.ArmorForgeRecipe;
+import com.mineboundteam.minebound.registry.MenuRegistry;
+import com.mineboundteam.minebound.registry.RecipeRegistry;
+import net.minecraft.network.FriendlyByteBuf;
+import net.minecraft.network.protocol.game.ClientboundContainerSetSlotPacket;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.world.Container;
+import net.minecraft.world.entity.player.Inventory;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.entity.player.StackedContents;
+import net.minecraft.world.inventory.*;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.crafting.Recipe;
+import net.minecraft.world.level.Level;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Optional;
+
+public class ArmorForgeMenu extends RecipeBookMenu<CraftingContainer> {
+    private final ContainerLevelAccess containerLevelAccess;
+    private final CraftingContainer craftingContainer = new CraftingContainer(this, 3, 3);
+    private final Player player;
+    private final ResultContainer resultContainer = new ResultContainer();
+    public static final int SIZE = 7;
+
+    public ArmorForgeMenu(int pContainerId, Inventory pPlayerInventory, FriendlyByteBuf friendlyByteBuf) {
+        this(pContainerId, pPlayerInventory, ContainerLevelAccess.NULL);
+    }
+
+    public ArmorForgeMenu(int containerID, Inventory inventory, ContainerLevelAccess containerLevelAccess) {
+        super(MenuRegistry.ARMOR_FORGE_MENU.get(), containerID);
+        this.containerLevelAccess = containerLevelAccess;
+        this.player = inventory.player;
+
+        for (int i = 0; i < 9; i++) {
```

Comment:
Please move slot registration into its own method so it can be overridden; once again that was the biggest headache with the base-game stuff